### DISCUSSION TITLE
PrimeNG Tree: How to allow exactly one selection?

### DISCUSSION THREAD

**Discussion by PranKe01** [primefaces/community#1646] [PrimeNG] [answered]
Hi there,
I have a tree and I want that exactly one item is selected. Not more not less. I am already using the selectionMode = single.
Anyhow, it is still possible to deselect that item, which I want to avoid. Unfortunately the onNodeUnselect emitter, does not contain an event, which I could cancel :/

Any ideas?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Unfortunately, this behavior is not supported. Also, I've checked now with v17.13.0 and see that the event is also emitted in onNodeUnselect emitter as `originalEvent: event`, but even if you try to cancel the event, it'll unselect the selection since the selection logic is not designed in that way.

**Reply by PranKe01** [reply]
Would it be possible to change that behaviour in future versions?
In the onClick function primeng could add a hook without much effort, I think?!
